lib/miq_automation_engine/models/miq_ae_domain.rb

```diff
@@ -45,6 +49,45 @@ def available_version
     version_field.try(:fetch_path, 'field', 'default_value')
   end
 
+  def self.import_git_repo(domain_name, git_repo, branch = nil, tag = nil)
+    MiqAeDomain.where(:name => domain_name).first.try(:destroy)
+    branch = 'origin/master' if branch.nil? && tag.nil?
+    options = {'git_dir' => git_repo.directory_name, 'branch' => branch, 'tag' => tag, 'preview' => false}
+    MiqAeImport.new(domain_name, options).import
+    domain = MiqAeDomain.where(:name => domain_name).first
+    raise MiqAeException::DomainNotFound, "Domain #{domain_name} not found after import" unless domain
+    domain.update_git_info(git_repo, branch, tag)
+  end
+
+  def update_git_info(git_repo, branch, tag)
+    self.branch = branch
+    self.git_repository = git_repo
+    self.tag = tag
+    save
```

Comment:
Why call save if you are going to call update_attributes immediately after?  Also, should it be `save!`?